Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -2320,24 +2357,30 @@ else if (startKind.equals("Location"))
 					throw new UserErrorMessage("You are not near this item, you cannot pick it up.");
 				
 				// Check if the character can actually carry something else or if its all too heavy...
-				Long itemWeight = getItemWeight(item);
+				// And if quantity is specified, measure for specified quantity rather than whole stack
+				Long itemWeight = (requestQuantity != null) ? requestQuantity*itemSingleWeight : getItemWeight(item);
+
+				// Again, not sure if this check is necessary, but leaving it for now
```

Comment:
I think it is necessary because getItemWeight() will probably return null if the weight of the item's weight is null, which can happen legitimately.